src/redata.js

```diff
@@ -73,21 +89,28 @@ const defaultInitialCtx = {
 
 const defaultInitialData = { loading: true, error: undefined, result: undefined };
 
+/**
+ * Calls the loader function with current params and returns a promise which resolves with the {@link data} format
+ * @param {function} loader - called to fetch new data
+ * @param {object} params - arguments object provided to loader function
+ * @param {function} onUpdate - called when new data arrives
+ * @return {Promise} - promise which resolves with the {@link data} format
+*/
 function load(loader, params, onUpdate) {
     // Init new data.
     const data = { ...defaultInitialData };
 
     // Start loading, passing the parameters that were provided, and return a promise for the loader resulting data.
     return loader(params)
-    .then((result) => { data.result = result; })
```

Comment:
I personally like to keep the indentation flat here. Not sure if there's an eslint rule to enforce this.